File: src/main/java/net/softbell/bsh/dto/view/admin/NodeManageItemCardDto.java
package net.softbell.bsh.dto.view.admin;

import lombok.Getter;
import lombok.Setter;
import net.softbell.bsh.domain.ItemCategoryRule;
import net.softbell.bsh.domain.ItemModeRule;
import net.softbell.bsh.domain.ItemTypeRule;
import net.softbell.bsh.domain.entity.NodeItem;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 노드 관리 수정뷰 아이템 카드정보 DTO
 */
@Getter
@Setter
public class NodeManageItemCardDto
{
	private long itemId;
	private String alias;
	private String itemName;
	private ItemModeRule itemMode;
	private ItemTypeRule itemType;
	private ItemCategoryRule itemCategory;
	private byte controlMode;
	
	public NodeManageItemCardDto(NodeItem entity)
	{
		// Exception
		if (entity == null)
			return;
		
		// Convert
		this.itemId = entity.getItemId();
		this.alias = entity.getAlias();
		this.itemName = entity.getItemName();
		this.itemMode = entity.getItemMode();
		this.itemType = entity.getItemType();
		this.itemCategory = entity.getItemCategory();
		this.controlMode = entity.getControlMode();
	}
}

File: src/main/kotlin/net/softbell/bsh/dto/view/admin/NodeManageItemCardDto.kt
package net.softbell.bsh.dto.view.admin

import net.softbell.bsh.domain.ItemCategoryRule
import net.softbell.bsh.domain.ItemModeRule
import net.softbell.bsh.domain.ItemTypeRule
import net.softbell.bsh.domain.entity.NodeItem

/**
 * @author : Bell(bell@softbell.net)
 * @description : 노드 관리 수정뷰 아이템 카드정보 DTO
 */
class NodeManageItemCardDto(entity: NodeItem) {
    val itemId: Long = entity.itemId
    val alias: String = entity.alias
    val itemName: String = entity.itemName
    val itemMode: ItemModeRule = entity.itemMode
    val itemType: ItemTypeRule = entity.itemType
    val itemCategory: ItemCategoryRule = entity.itemCategory
    val controlMode: Byte = entity.controlMode
}